### PR TITLE
the limits specifies before nullability.

### LINKED ISSUE #9291: CDS reader ignores nullability for columns with a limits specifier
The CDS reader fails to treat columns as nullable if the ReadMe file contains a limits specifier. Examples include the `Tc`, `Lc` and `phi` columns of the main.dat file in [XHIP (V/137D)](http://cdsarc.u-strasbg.fr/ftp/V/137D/ReadMe)

The issue appears to be the following regular expression: https://github.com/astropy/astropy/blob/af4708ec1aedccdd1fb304704a59fcb7f29ec567/astropy/io/ascii/cds.py#L128-L129

 This will only match a `?` character at the start of the description. As the [file format description](http://vizier.u-strasbg.fr/doc/catstd-3.4.htx) states, the limits must be specified *before* nullability.

As a local fix, I have replaced the regular expression with

    r'(?:[\[\]] \S* [\[\]])? \? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?'

This handles the case where limits are supplied. I'm not sure either this or the current version will work properly in the presence of order specifiers though.

### PR DESCRIPTION
**Pull request by NikitaTewary**
### Description
The following changes are just a modification for CDS reader ignores nullability for columns with a limits specifier, since it was not working in 2019, the following code may work now.
You can add the io.ascii label to it.

Fixes #9291

### CODE REVIEW

**Comment by hamogu** [comment]
Hi @NikitaTewary great that you open a PR here. You'll see that you code failed all tests because your changes broke the code. Your lines does not even close the ")". You probably did not intend to delete the line below?
Always run the tests locally to check for those things before you push to github!

When that is fixed, you should add a test. Look at the existing tests for the cds module as an example. Maybe you can modify one of the existing tables in the /test/data directory to test this case, too?

File: astropy/io/ascii/cds.py [hunk 1]
```diff
@@ -126,8 +126,7 @@ def get_cols(self, lines):
                 col.type = self.get_col_type(col)
 
                 match = re.match(
-                    r'\? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?',
-                    col.description, re.VERBOSE)
```
Comment (hamogu) [inline astropy/io/ascii/cds.py:130]: Is there a specific reason that you removed this line? Did you run the tests locally before pushing the code to github?

Comment (NikitaTewary) [reply astropy/io/ascii/cds.py:130]: As stated in the #9291 issue, i also felt the best way to fix the nullability problem was to change 
r'\? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?', col.description, re.VERBOSE) 
this line to
r'(?:[\[\]] \S* [\[\]])? \? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?'

Comment (hamogu) [reply astropy/io/ascii/cds.py:130]: Well, it won't work then. How is this supposed to work? Look up the calling signature of "re.match()" and you'll see that you need to pass in a string that it matches **to**. Please use your own Python programming skills and don't just copy a hint that someone gave in #9291 to copy and paste exactly. That was just that - a hint how to make it work, not a complete programming solution.
And please run the tests locally and make sure they pass (that don't with this!) before you push to github.

Comment (NikitaTewary) [reply astropy/io/ascii/cds.py:130]: Okay got it, it won't happen again sorry for the inconvenience caused!

**Comment by pllim** [comment]
Hello and thank you! I see that this is still work in progress, so I took the liberty to turn this into a draft PR. When this is ready, feel free to remove the draft status.

Hopefully our [developer documentation](https://docs.astropy.org/en/latest/#developer-documentation) would explain the workflow enough for you to proceed.

**Pull request closed without merging**